app/controllers/organization_alliances_controller.rb

```diff
@@ -0,0 +1,79 @@
+class OrganizationAlliancesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :member_should_exist_and_be_active
+  before_action :authorize_admin
+  before_action :find_alliance, only: [:update, :destroy]
+
+  def index
+    @status = params[:status] || "pending"
+
+    @alliances = case @status
+                 when "pending"
+                   current_organization.pending_sent_alliances.includes(:source_organization, :target_organization) +
+                   current_organization.pending_received_alliances.includes(:source_organization, :target_organization)
+                 when "accepted"
+                   current_organization.accepted_alliances.includes(:source_organization, :target_organization)
+                 when "rejected"
+                   current_organization.rejected_alliances.includes(:source_organization, :target_organization)
+                 else
+                   []
+                 end
+  end
+
+  def create
+    @alliance = OrganizationAlliance.new(
+      source_organization: current_organization,
+      target_organization_id: params[:organization_alliance][:target_organization_id],
+      status: "pending"
+    )
+
+    if @alliance.save
+      flash[:notice] = t("organization_alliances.created")
+    else
+      flash[:error] = @alliance.errors.full_messages.to_sentence
+    end
+
+    redirect_back fallback_location: organizations_path
+  end
+
+  def update
+    authorize @alliance
```

Comment:
since we have `before_action :authorize_admin` we don't need an `authorize`  here